compiler/noirc_frontend/src/hir/comptime/hir_to_ast.rs

```diff
@@ -0,0 +1,345 @@
+use iter_extended::vecmap;
+use noirc_errors::{Span, Spanned};
+
+use crate::ast::{ConstrainStatement, Expression, Statement, StatementKind};
+use crate::hir_def::expr::{HirArrayLiteral, HirExpression, HirIdent};
+use crate::hir_def::stmt::{HirLValue, HirPattern, HirStatement};
+use crate::macros_api::HirLiteral;
+use crate::node_interner::{ExprId, NodeInterner, StmtId};
+use crate::{
+    ArrayLiteral, AssignStatement, BlockExpression, CallExpression, CastExpression, ConstrainKind,
+    ConstructorExpression, ExpressionKind, ForLoopStatement, ForRange, Ident, IfExpression,
+    IndexExpression, InfixExpression, LValue, Lambda, LetStatement, Literal,
+    MemberAccessExpression, MethodCallExpression, Path, Pattern, PrefixExpression, Type,
+    UnresolvedType, UnresolvedTypeData, UnresolvedTypeExpression,
+};
+
+// TODO:
+// - Full path for idents & types
+// - Assert/AssertEq information lost
+// - The type name span is lost in constructor patterns & expressions
+// - All type spans are lost
+// - Type::TypeVariable has no equivalent in the Ast
+
+impl StmtId {
+    #[allow(unused)]
+    fn to_ast(self, interner: &NodeInterner) -> Statement {
+        let statement = interner.statement(&self);
+        let span = interner.statement_span(&self);
+
+        let kind = match statement {
+            HirStatement::Let(let_stmt) => {
+                let pattern = let_stmt.pattern.into_ast(interner);
+                let r#type = interner.id_type(let_stmt.expression).to_ast();
+                let expression = let_stmt.expression.to_ast(interner);
+                StatementKind::Let(LetStatement { pattern, r#type, expression })
+            }
+            HirStatement::Constrain(constrain) => {
+                let expr = constrain.0.to_ast(interner);
+                let message = constrain.2.map(|message| message.to_ast(interner));
+
+                // TODO: Find difference in usage between Assert & AssertEq
+                StatementKind::Constrain(ConstrainStatement(expr, message, ConstrainKind::Assert))
```

Comment:
where does the second `ValueId` in the `Constrain` end up? i.e. `constrain.1`

Reply:
`constrain.1` is a FileId, not a ValueId